elements/pl-external-grader-variables/pl-external-grader-variables.py

```diff
@@ -1,33 +1,90 @@
-import prairielearn as pl
-import lxml.html
 import chevron
+import lxml
+import prairielearn as pl
 
-
-PARAMS_NAME_DEFAULT = None
+EMPTY_DEFAULT = False
 
 
-def prepare(element_html, element_index, data):
+def prepare(element_html: str, data: pl.QuestionData) -> None:
     element = lxml.html.fragment_fromstring(element_html)
-    pl.check_attribs(element, ['params-name'], [])
+    pl.check_attribs(element, ["params-name"], ["empty"])
+
+    params_name = pl.get_string_attrib(element, "params-name")
+
+    # Get frontend variables
+    frontend_variables = []
```

Comment:
What does "frontend" mean in this context? Is this referring to the fact that this data comes from `question.html`? AFAIK we don't use that terminology elsewhere; maybe name this `html_variables` instead?

Reply:
Yes, taking variables from HTML is the intended meaning. Will change.